### ISSUE TITLE
Map specific version of vulnerability as affected vulnerable software

### ISSUE THREAD

**Issue by sahibamittal** [DependencyTrack/hyades#606]
Currently, vulnerabilities are missing affected package version info due to indefinite ranges. 
Example, for a vulnerability with version ">7.0", as per previous discussions, we decided to skip mapping such open ranges.

Proposal: When we search for vulnerability based on a purl with specific version and say, SNYK reports a vulnerability for it, we can at least create a vulnerable software for the specific version, since it is affected indeed.
This will keep a record of all definite affected versions for a vulnerability.

**Comment by nscuro** [comment]
IIRC we do not ingest any affected versions into the DB, if the data comes from sources that we only use for ad-hoc analysis (OSS Index, Snyk). Reason being that the danger of inconsistencies arising is quite high.

Remember that everything we ingest will be picked up by the internal analyzer. It is thus possible that the internal analyzer operates based on stale data, that does not reflect what the external source (OSS Index) would report.

What is the exact use case behind this enhancement? Is it just to display the versions in the UI?

**Comment by sahibamittal** [comment]
> IIRC we do not ingest any affected versions into the DB, if the data comes from sources that we only use for ad-hoc analysis (OSS Index, Snyk). Reason being that the danger of inconsistencies arising is quite high.
> 
> Remember that everything we ingest will be picked up by the internal analyzer. It is thus possible that the internal analyzer operates based on stale data, that does not reflect what the external source (OSS Index) would report.
> 
> What is the exact use case behind this enhancement? Is it just to display the versions in the UI?

Yes, example : if SNYK reports a vulnerability (for a component version 9.0) with affected version range '>7.0', we don't parse it (open ended range) and this info is not stored/displayed. So idea is to document at least the specific affected component version (9.0) in this case if not parsing the indefinite range.

**Comment by sahibamittal** [comment]
Upon discussion, there are still challenges for this implementation. Like, one CVE may affect multiple components and for one component, we store only one specific version (not range), the data may be incomplete and inconsistent. Internal analyzer uses the vulnerability data (fetched from mirror and scanners) and this may cause inconsistency.

If the range data is required just for display, we can store it separately but is it even worth the complexity? This will impact performance and storage.

**Comment by nscuro** [comment]
No longer relevant. I don't remember when this changed, but we do store and evaluate open-ended ranges now. I believe the original concern was mostly due to unreliable data from OSV, and findings never being removed once they were added.

As of https://github.com/DependencyTrack/dependency-track/blob/main/docs/adr/013-finding-status.md we deactivate findings that are no longer reported, so we handle data correction from upstream sources better than we did when this issue was raised.
